packages/astro/src/@types/astro.ts

```diff
@@ -705,6 +705,23 @@ export interface AstroUserConfig {
 	 */
 	cacheDir?: string;
 
+	/**
+	 * @docs
+	 * @name typegenDir
+	 * @type {string}
+	 * @default `"./.astro"`
+	 * @description TODO
+	 *
+	 * The value can be either an absolute file system path or a path relative to the project root.
+	 *
+	 * ```js
+	 * {
+	 *   typegenDir: './my-custom-cache-directory'
+	 * }
+	 * ```
+	 */
+	typegenDir?: string;
```

Comment:
It'd be totally okay with this not being customizable if it makes the logic more complex everywhere, fwiw.